CMakeLists.txt

```diff
@@ -212,17 +215,23 @@ if("${ENABLE_ALL_WARNINGS}")
          -Wno-strict-aliasing \
          -Wno-type-limits")
   endif()
+  if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+    set(KNOWN_SYSTEM_SPECIFIC_WARNINGS "-Wno-nullability-completeness")
```

Comment:
What does this flag do ?

Reply:
The issue seems to be replicated also on this PR: https://github.com/facebookincubator/velox/pull/2608 as seen on this job:
https://app.circleci.com/pipelines/github/facebookincubator/velox/13795/workflows/d149e026-0c44-4555-ae47-81359439ffee/jobs/83267
```
FAILED: velox/exec/CMakeFiles/velox_exec.dir/GroupingSet.cpp.o 
ccache /Applications/Xcode-12.5.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DFMT_LOCALE -DGFLAGS_IS_A_DLL=0 -I/usr/local/opt/icu4c/include -I/Users/distiller/project/. -I/Users/distiller/project/velox/external/xxhash -I/Users/distiller/project/third_party/xsimd/include -isystem /Users/distiller/deps/include -isystem /Users/distiller/project/velox -isystem /Users/distiller/project/velox/external -isystem /Users/distiller/project/velox/external/duckdb -isystem /Users/distiller/project/velox/external/duckdb/tpch/dbgen/include -isystem /Users/distiller/deps/opt/openssl@1.1/include -isystem /Users/distiller/project/third_party/googletest/googletest/include -isystem /Users/distiller/project/third_party/googletest/googletest -mavx2 -mfma -mavx -mf16c -mlzcnt -std=c++17 -mbmi2 -fvisibility=hidden -fvisibility-inlines-hidden -D USE_VELOX_COMMON_BASE -D HAS_UNCAUGHT_EXCEPTIONS -Werror -Wall -Wextra -Wno-unused        -Wno-unused-parameter        -Wno-sign-compare        -Wno-ignored-qualifiers        -Wno-range-loop-analysis -Wno-mismatched-tags -g -isysroot /Applications/Xcode-12.5.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -std=gnu++17 -MD -MT velox/exec/CMakeFiles/velox_exec.dir/GroupingSet.cpp.o -MF velox/exec/CMakeFiles/velox_exec.dir/GroupingSet.cpp.o.d -o velox/exec/CMakeFiles/velox_exec.dir/GroupingSet.cpp.o -c /Users/distiller/project/velox/exec/GroupingSet.cpp
In file included from /Users/distiller/project/velox/exec/GroupingSet.cpp:16:
In file included from /Users/distiller/project/./velox/exec/GroupingSet.h:19:
/Users/distiller/project/./velox/exec/HashTable.h:83:36: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
    const raw_vector<vector_size_t>* rows{nullptr};
                                   ^
/Users/distiller/project/./velox/exec/HashTable.h:83:36: note: insert '_Nullable' if the pointer may be null
    const raw_vector<vector_size_t>* rows{nullptr}
```
I am slightly confused as I am not sure why is not failing on other PRs too. I thought the issue was related to the folly move but seems to happen on other PRs.

Reply:
I guess some recent PRs might have fixed this problem : https://github.com/facebookincubator/velox/pull/2593 . Does updating to latest in main without this flag work ?

Reply:
Thanks, it seems this is not required outside from some folly specific warnings supressions now.